src/engine/SpatialJoinConfig.h

```diff
@@ -30,6 +30,11 @@ enum class SpatialJoinType {
   WITHIN_DIST
 };
 
+// String representation of the `SpatialJoinType` values.
+static constexpr std::string_view SpatialJoinTypeString[8]{
+    "intersects", "contains", "covers",   "crosses",
+    "touches",    "equals",   "overlaps", "within-dist"};
+
```

Comment:
I agree with SonarCloud that this should be an `std::array`. Or is there a problem with that?